enhancements/security/tls-artifacts-registry.md

```diff
@@ -0,0 +1,308 @@
+---
+title: tls-artifacts-registry
+authors:
+  - "@vrutkovs"
+  - "@dgrisonnet"
+  - "@dinhxuanvu"
+reviewers:
+  - "@deads2k"
+approvers:
+  - "@deads2k"
+api-approvers:
+  - "@deads2k"
+creation-date: 2023-10-26
+last-updated: 2023-12-08
+tracking-link:
+  - https://issues.redhat.com/browse/API-1603
+---
+
+# TLS Artifacts Registry
+
+## Summary
+
+This enhancement describes a single registry to store all produced 
+TLS artifacts. Both engineering and customers have expressed concern 
+about a lack of understanding what is the purpose of a particular 
+certificate or CA bundle, who issued it, and which team is responsible 
+for its contents and lifecycle. This enhancement describes a registry 
+to store necessary information about all certificates in the cluster 
+and sets requirements for TLS artifacts metadata.
+
+## Motivation
+
+TLS artifacts registry is useful for several teams in Red Hat (i.e. 
+which component manages a particular certificate) and customers - 
+for instance, helping them to separate platform-managed certificates 
+from user-managed.
+
+### User Stories
+
+- As an Openshift administrator, I want to differentiate 
+  platform-managed TLS artifacts from user-managed.
+- As an Openshift developer, I want to be able to find which team 
+  manages a particular certificate.
+- As an Openshift developer, I want to have a test which verifies 
+  that all certificates managed by the team have sufficient metadata
+  to explain its purpose
+- As a member of OpenShift concerned with the release process 
+  (TRT, dev, staff engineer, maybe even PM), I want to make sure all 
+  managed certificates have sufficient metadata to find the team 
+  responsible for its lifecycle.
+
+### Goals
+
+Create a registry of all managed TLS artifacts, and ensure that no new 
+certificates are added without sufficient metadata.
+Add a test to the conformance suite that verifies the actual cluster TLS 
+artifacts have necessary metadata.
+
+### Non-Goals
```

Comment:
I'd call out here the bits from earlier comments, ie. JWT tokens and cluster external certificates.

Reply:
Good idea, added few items here